doc/man7/EVP_PKEY-ML-KEM.pod

```diff
@@ -33,48 +35,93 @@ Generated keys default to retaining the seed used.
+
+In addition to the common parameters that all keytypes should support (see
+L<provider-keymgmt(7)/Common Information Parameters>), B<ML-KEM> keys
+keys support the following.
+
+=over 4
+
+=item "pub" (B<OSSL_PKEY_PARAM_PUB_KEY>) <octet string>
+
+The public key value.
+
+This parameter is used when importing or exporting the public key value with
+the EVP_PKEY_fromdata() and EVP_PKEY_todata() functions.
+The value format is that of the FIPS 203 (Algorithm 16:
+B<ML-KEM.KeyGen_internal>) B<ek> public key.
+The key length and content is that of the FIPS 203 (Algorithm 16:
+B<ML-KEM.KeyGen_internal>) B<ek> public key for the given ML-KEM variant.
+Initial import aside, this parameter is otherwise only gettable.
+
+=item "priv" (B<OSSL_PKEY_PARAM_PRIV_KEY>) <octet string>
+
+The private key value.
+
+This parameter is used when importing or exporting the private key value with
+the EVP_PKEY_fromdata() and EVP_PKEY_todata() functions.
+The key length and content is that of the FIPS 203 (Algorithm 16:
+B<ML-KEM.KeyGen_internal>) B<dk> private key for the given ML-KEM variant.
+Initial import aside, this parameter is otherwise only gettable.
+
+=item "encoded-pub-key" (B<OSSL_PKEY_PARAM_ENCODED_PUBLIC_KEY>) <octet string>
+
+Used for getting and setting the encoding of a public key.
+The key format is that of B<ek> in FIPS 203, Algorithm 16:
+B<ML-KEM.KeyGen_internal>.
+Updates of the public and private key components are only allowed on keys that
+are empty.
+Once a public or private key component is set, no further changes are allowed.
+This parameter is gettable and settable (once only).
+
+=back
+
+=head2 Provider configuration parameters
 
-Retention of the seed can be disabled by setting the C<pkey_seed_retain>
-parameter of the C<default> provider to C<no>.
 See the description of the B<-provparam> option in L<openssl(1)> to learn
-how to set provider options in the command line tools.
+how to set provider configuration parameters in the command line tools.
 See L<OSSL_PROVIDER_add_conf_parameter(3)> to learn how to set provider
 configuration options programmatically.
-When using the FIPS provider, the parameter may need to be set in the C<base>
-provider instead (or perhaps in all providers) if the intent is to retain seeds
-read from PKCS#8 files.
 
-In addition to its use in key generation, this parameter is also gettable and
-settable.
+=over 4
+
+=item C<ml-kem.retain_seed> (B<OSSL_PKEY_PARAM_ML_KEM_RETAIN_SEED>)
 
-See L<provider-keymgmt(7)/Common Information Parameters> for further
-information.
+When set to a string representing a false boolean value (see
+L<OSSL_PROVIDER_conf_get_bool(3)>), the seed will not be retained after key
+generation or key import from a seed value.
+If the resulting key is then written to a PKCS#8 object, it will contain
+only the FIPS 203 C<dk> key.
 
-=item C<ml-kem.retain_seed> (B<OSSL_PKEY_PARAM_ML_KEM_RETAIN_SEED>) <int>
+=item C<ml-kem.prefer_seed> (B<OSSL_PKEY_PARAM_ML_KEM_PREFER_SEED>)
```

Comment:
?

Reply:
These are provider config parameters, not used as regular params otherwise.

Reply:
They are presented as `OSSL_PARAM` items, are they not?  And they can also be set by the application, can they not?  Either way, it's not just an indicator for the application, it's also an indicator for providers that want to support this particular parameter.

Reply:
Well, these parameters are not used in param arrays by any providers, they're only queried as parameter config variables, which are always UTF8_STRING, but I can put those type back, if you feel they're important. No big deal.

Reply:
Types restored.